### PR TITLE
emit success event with subscription id

### PR DESCRIPTION
**Pull request by davidleger95**
## Change

- added a success event for when a subscription succeeds.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1 +1,9 @@
 export * from './components';
+
+const createElement = document.querySelector('manifold-subscription-create');
+
+// eslint-disable-next-line no-unused-expressions
+createElement?.addEventListener<any>('onSuccess', (e: any) => {
+  console.log(e);
+  alert(e.detail.id);
+});
```
Comment (davidleger95) [inline src/index.ts:9]: This is just to test out the event. It doesn't seem to be working. I've tried `success` `onSuccess`

File: src/components/manifold-subscription-create/manifold-subscription-create.tsx [hunk 2]
```diff
@@ -205,36 +215,45 @@ export class ManifoldSubscriptionCreate {
...
+        if (data) {
+          console.log(data);
+          this.createSuccess.emit({ id: data.createSubscription.data.id });
         }
       }
+      this.subscribing = false;
     }
   };
 
+  @Event({ eventName: 'success' }) createSuccess: EventEmitter<{ id: string }>;
```
Comment (davidleger95) [inline src/components/manifold-subscription-create/manifold-subscription-create.tsx:255]: might need a specific flag here to expose the event

File: src/components/manifold-subscription-create/manifold-subscription-create.tsx [hunk 3]
```diff
@@ -205,36 +215,45 @@ export class ManifoldSubscriptionCreate {
...
+          variables,
+        });
+
+        if (errors) {
+          // TODO handle errors
+        }
+
+        if (data) {
+          console.log(data);
+          this.createSuccess.emit({ id: data.createSubscription.data.id });
```
Comment (dangodev) [inline src/components/manifold-subscription-create/manifold-subscription-create.tsx:248]: Does this do anything to fire the event?

```diff
- async fetchCustomCost(configuredFeatures: FeatureMap) {
+fetchCustomCost: async (configuredFeatures: FeatureMap) => {
```

Comment (dangodev) [reply src/components/manifold-subscription-create/manifold-subscription-create.tsx:248]: Oh wait sorry; this is in a different function (I misread the diff). I don’t see anything apparently off with this code; seems like it should work

File: src/components/manifold-subscription-create/manifold-subscription-create.tsx (hunk 2 above)
Comment (sslotsky) [reply src/components/manifold-subscription-create/manifold-subscription-create.tsx:255]: The event fires for me

```
    <script>
      const create = document.querySelector('manifold-subscription-create');
      create.addEventListener('success', console.log);
    </script>
```

File: src/components/manifold-subscription-create/manifold-subscription-create.tsx (hunk 3 above)
Comment (dangodev) [reply src/components/manifold-subscription-create/manifold-subscription-create.tsx:248]: Oh maybe it’s the `onSubmit={this.subscribe}`. Does `onSubmit{(evt) => this.subscribe(evt)}` do anything differently?

Comment (davidleger95) [reply src/components/manifold-subscription-create/manifold-subscription-create.tsx:247]: It's was all working correctly. Sam's solution works. I think it was adding the event listener too early before the element was added to the dom

File: src/index.html [hunk 4]
```diff
@@ -31,7 +31,11 @@
     <script nomodule src="/build/manifold-subscription.js"></script>
   </head>
   <body>
-    <manifold-init env="stage" client-id="3zx3qeb15z70kcv4nh8cjedhndwz2"></manifold-init>
+    <manifold-init
+      env="stage"
+      client-id="3zx3qeb15z70kcv4nh8cjedhndwz2"
+      auth-token="eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiIyMWpkZmhyenVqa2U0emVldmVyMTZ6eWRjNXhteSIsInN1YiI6Im1vdGhlck9EcmFnb25zIiwiaWF0IjoiMjAyMC0wNC0xNVQxNDo0Mjo1OS40MzQ5ODIxOTdaIiwiZXhwIjoiMjAyMC0wNC0xNlQxNDo0Mjo1OS40MzQ5ODIxOTdaIiwiYXVkIjoibWFuaWZvbGQuY28vZ2F0ZWtlZXBlciIsInZhbHVlIjoiVmRxbUFUTzgiLCJwbGF0Zm9ybSI6IjN6eDc3ZHVxamZ3NGpoZXBmajMydTR2MnJubTFlIiwidG9rZW5fc2VjcmV0Ijoid2lxYlNGclVZOTE5WUN1dGhrUE42N3VxcnFTcVpLbTFzX2ZvOG1CUlhNLWFiVi1CeG42TkRleXFUVThYIn0.bq2e41Hngrq7lUSMOsSGEJwWWfUz9-vS-XUurYIGL-SuLKQUWCn_wPXoc0BkgY5SdW440L6mJXVef7H8CHI9Mw"
```
Comment (dangodev) [inline src/index.html:37]: This was good for testing (I saw it working myself), but do we want to leave this here?

File: src/components/manifold-subscription-create/manifold-subscription-create.tsx [hunk 5]
```diff
@@ -205,36 +215,44 @@ export class ManifoldSubscriptionCreate {
...
+
+        if (errors) {
+          // TODO handle errors
+        }
+
+        if (data) {
+          this.createSuccess.emit({ id: data.createSubscription.data.id });
         }
       }
+      this.subscribing = false;
```
Comment (dangodev) [inline src/components/manifold-subscription-create/manifold-subscription-create.tsx:250]: Just a note: when I tried to subscribe the 2nd time, nothing happened—no error state or anything. Should there be some “finish” state when successful (if yes, probably not a part of this PR, but still wanted to ask)?

Comment (davidleger95) [reply src/components/manifold-subscription-create/manifold-subscription-create.tsx:250]: We decided to just let the developer handle the success state by listening for the event. I think it should probably reset though and allow another subscription once it's done with the first one

**Pull request merged**